Automattic/jetpack
# Launchpad: New update_about_page task

Fixes https://github.com/Automattic/wp-calypso/issues/78951

## Proposed changes:

This adds a new 'Update About page' Launchpad task on `/home`. The task is only visible if the site has an 'About' page that was headstarted with the theme. In the event that there isn't an 'About' page, we fall back to the 'Add new page' and 'Edit page' tasks.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






### Testing Setup

* Apply this PR to Jetpack on your sandbox and sandbox the public-api.
* Run the calypso branch from [79069](https://github.com/Automattic/wp-calypso/pull/79069) locally or use the calypso.live link. This ensures the new Launchpad task will be clickable

### English site testing

* Make sure you WordPress.com account locale is set to English.
* Go to `/start` and create a new site. Using the Free plan is fine.
* Select the "Promote myself or business" goals/intent.
* Select a theme that headstarts an 'About' page (Barnsbury23 is a good example, see 31077-pb/#js for other possibilities).
* Launch the site.
* When you reach the dashboard, you should see an 'Update About page' task.
![image](https://github.com/Automattic/jetpack/assets/917632/7dff4a27-7119-40c8-9818-1efbde47e39c)
* Clicking the task should take you to the edit view for the 'About' page. 
* Make an edit to the page and publish it. The task should be marked complete when you return to the dashboard.

### Non-English site testing

**NOTE: The task will not appear until the 'Update your About page' string has been translated.**

* Make sure you WordPress.com account locale is set to another language. I've been testing with Spanish (`es`).
* Go to `/start` and create a new site. Using the Free plan is fine.
* Select the "Promote myself or business" goals/intent.
* Select a theme that headstarts an 'About' page (Barnsbury23 is a good example, see 31077-pb/#js for other possibilities).
* Launch the site.
* When you reach the dashboard, you should see an 'Update your About page' task.
* Clicking the task should take you to the edit view for the 'About' page in the correct language. 
* Make an edit to the page and publish it. The task should be marked complete when you return to the dashboard.

### Testing the 'Add a new page' task

* On either of the two test sites you created above, go to the dashboard and click the 'Add new page' task.
![image](https://github.com/Automattic/jetpack/assets/917632/f8ca33d5-92b2-4aed-b894-e897c83f9cf3)
* Create a new page of any type (other than 'About') and publish it.
* When you return to the dashboard, the 'Add new page' task should be marked complete and the 'Edit page' task _should not_ be visible. The 'Update About' task should still be there.
![image](https://github.com/Automattic/jetpack/assets/917632/b4d2ec54-c8ea-4186-b846-59cd70f1090d)

### Testing a site without an 'About' page
* Go to `/start` and create a new site. Using the Free plan is fine.
* Select the "Promote myself or business" goals/intent.
* Select a theme that _does not_ headstart an 'About' page (Russell is a good example).
* Launch the site.
* When you return to the dashboard, you should see the 'Add new page' task but _not_ the 'Update About page' task.
![image](https://github.com/Automattic/jetpack/assets/917632/67caa115-9bea-4863-bc53-d91dab3c6c28)
* Click the 'Add new page' task.
* Create a new page of any type (other than 'About') and publish it.
* When you return to the dashboard, the 'Add new page' task should be marked complete and the 'Edit page' task _should_ be visible.
![image](https://github.com/Automattic/jetpack/assets/917632/58b74050-6ccd-439a-b63f-da6c960f9672)
